### PR TITLE
feat(ticketing): retrieve existing tickets from storage

### LINKED ISSUE #477: [TICKETING] Retrieve existing tickets from storage
Implement the method required to retrieve user tickets in the infrastructure adapter, fulfilling the contract defined in the output port.

Only the empty-list case is covered in this PR.
Testing actual retrieval by user depends on having ticket persistence populated through the corresponding ticket creation flow, which is being handled in a separate issue.

## Acceptance Criteria

- [ ] The adapter implements the output port method.
- [ ] The adapter correctly returns the existing tickets for the given user.
- [ ] A simple test is included.
- [ ] No endpoint logic is added in this PR.

### PR DESCRIPTION
**Pull request by JordiCasas87**
Closes #477

## Summary
- implement `findAllByUserId(String userId)` in `InMemoryTicketRepository`
- fulfill the contract defined in `TicketRepository`
- add a simple test covering the empty-list case

## Note
This PR only covers the storage-side retrieval logic for user tickets and does not include endpoint logic.

Only the empty-list case is covered in this PR. Testing actual retrieval by user depends on having ticket persistence populated through the corresponding ticket creation flow, which is being handled in a separate issue.

### CODE REVIEW

**Review by eariasvalor — APPROVED** [review]
Everything looking good!

**Review by andyrouge81 — APPROVED** [review]
GJ! Clean and well-scoped PR aligned with issue #477.

I like that the implementation stays focused only on the repository/storage responsibility without mixing endpoint logic. Everything looks clear and easy to follow.

Just one small question: was the decision to keep the `storage` field without `private` intentional for this MVP phase, maybe to facilitate future tests or allow faster package access while the ticketing flow is still evolving?

Apart from that, GJ!

**Comment by JordiCasas87** [comment]
@andyrouge81  Good catch, thanks Andrés. I’ve updated it so `storage` is now `private final`.

File: backend/account-service/src/main/java/com/ita/challenges/account/infrastructure/adapter/web/out/persistence/InMemoryTicketRepository.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
 @Repository
 public class InMemoryTicketRepository implements TicketRepository {
 
-    Map<String, Ticket> storage = new ConcurrentHashMap<>();
+    private final Map<String, Ticket> storage = new ConcurrentHashMap<>();
```
Comment (andyrouge81) [inline backend/account-service/src/main/java/com/ita/challenges/account/infrastructure/adapter/web/out/persistence/InMemoryTicketRepository.java:15]: Thanks for applying the requested changes! The encapsulation with `private final` looks cleaner and more controlled now. Hopefully it will not cause the previously discussed access or testing issues, but overall it seems like a good improvement. GJ!

**Review by AlbertMedina — APPROVED** [review]
Good job on the implementation. LGTM!

**Review by kenan-it-academy — APPROVED** [review]
